### PR TITLE
PS-5188 : Cannot allocate temporary tablespace for current session if

### PR DESCRIPTION
**Pull request by satya-bodapati**
innodb_encrypt_tables = keyring_on | keyring_force | online_to_keyring |
          online_to_keyring_force | online_from_keyring_to_unencrypted

Problem:
--------
Keyring encryption modes are not handled for session temporary tablespaces

Fix:
----
Keyring encryption is neither applicable for global temporary tablespace (ibtmp1)
or session temporary tablespaces (*.ibt).

We use master key encrpytion for temp tablespaces when keyring encryption is enabled.
i.e. for these keyring encryption modes:

  SRV_ENCRYPT_TABLES_KEYRING_ON = 3,
  SRV_ENCRYPT_TABLES_KEYRING_FORCE = 4,
  SRV_ENCRYPT_TABLES_ONLINE_TO_KEYRING = 5,
  SRV_ENCRYPT_TABLES_ONLINE_TO_KEYRING_FORCE = 6,
  SRV_ENCRYPT_TABLES_ONLINE_FROM_KEYRING_TO_UNENCRYPTED = 7

innodb.innodb_scrub_background test now fails with "result file doesn't exist"

### CODE REVIEW

File: storage/innobase/dict/dict0crea.cc [hunk 1]
```diff
@@ -186,9 +186,27 @@ dberr_t dict_build_tablespace(
...
-      srv_tmp_tablespace_encrypt) {
+  bool encrypted = false;
+  switch (srv_encrypt_tables) {
+    case SRV_ENCRYPT_TABLES_ON:
+    case SRV_ENCRYPT_TABLES_FORCE:
+    case SRV_ENCRYPT_TABLES_KEYRING_ON:
+    case SRV_ENCRYPT_TABLES_KEYRING_FORCE:
+    case SRV_ENCRYPT_TABLES_ONLINE_TO_KEYRING:
+    case SRV_ENCRYPT_TABLES_ONLINE_TO_KEYRING_FORCE:
+    case SRV_ENCRYPT_TABLES_ONLINE_FROM_KEYRING_TO_UNENCRYPTED:
```
Comment (robgolebiowski) [inline storage/innobase/dict/dict0crea.cc]: for TO_UNECRYPTED - table should not be encrypted.

**Review by robgolebiowski — CHANGES_REQUESTED** [review]

**Comment by satya-bodapati** [comment]
addressed comments from @robgolebiowski

File: storage/innobase/dict/dict0crea.cc [hunk 2]
```diff
@@ -186,9 +186,27 @@ dberr_t dict_build_tablespace(
...
-      srv_tmp_tablespace_encrypt) {
+  bool encrypted = false;
+  switch (srv_encrypt_tables) {
+    case SRV_ENCRYPT_TABLES_ON:
+    case SRV_ENCRYPT_TABLES_FORCE:
+    case SRV_ENCRYPT_TABLES_KEYRING_ON:
+    case SRV_ENCRYPT_TABLES_KEYRING_FORCE:
+    case SRV_ENCRYPT_TABLES_ONLINE_TO_KEYRING:
+    case SRV_ENCRYPT_TABLES_ONLINE_TO_KEYRING_FORCE:
+      encrypted = true;
```
Comment (robgolebiowski) [inline storage/innobase/dict/dict0crea.cc:197]: does not srv_tmp_talblespace_encrypt be set to true in order to set "encrypted = true" ? Is it validated somewhere ?

Comment (robgolebiowski) [reply storage/innobase/dict/dict0crea.cc:197]: or this encrypted - means temp tablespace encrypted ?

Comment (robgolebiowski) [reply storage/innobase/dict/dict0crea.cc:197]: but then should not encrypted=true be set if srv_tmp_tablespace_encrypt is set ?

Comment (satya-bodapati) [reply storage/innobase/dict/dict0crea.cc:197]: It is bool variable. Regarding validation, it Is done. Encryption::check_keyring. See the code around fil_temp_update_encryption

**Review by robgolebiowski — APPROVED** [review]

**Pull request merged**